metadataproxy/roles.py

```diff
@@ -86,6 +86,34 @@ def sts_client():
     return _sts_client
 
 
+@log_exec_time
+def inspect_container_by_id(container_id):
+    client = docker_client()
+    with PrintingBlockTimer('Container inspect'):
+        return client.inspect_container(container_id)
+
+
+@log_exec_time
+def find_container_by_network(ip, network_name):
+    client = docker_client()
+    network = None
+    with PrintingBlockTimer('Network inspect'):
+        try:
+            network = client.inspect_network(network_name)
```

Comment:
what is the cost of this operation if you have 1000s of containers running? Will it return all the containers?

Reply:
Method `inspect_network` makes an HTTP GET request to the docker daemon in the host.  
We could cache the response but the call happens once per container.  It would be a problem if many containers asked for credentials for the first time at once.

Reply:
I don't think we should be caching it because new container are added all the time